src/mame/drivers/myb3k.cpp

```diff
@@ -77,173 +91,426 @@ class myb3k_state : public driver_device
+
+					for (int pxl = 0; pxl < 8; pxl++)
+					{
+						if ((pdat & (0x80 >> pxl)) != 0)
+						{
+							bitmap.pix32(y, ( x_pos * 8) + pxl) = (*m_pal)[0x07];
+						}
+						else
+						{
+							bitmap.pix32(y, ( x_pos * 8) + pxl) = rgb_t::black();
+						}
+					}
 				}
+				break;
+			default:
+				logerror("unimplemented video mode: %02x\n", m_vmode);
 			}
 		}
 	}
-
-	return 0;
-}
-
-WRITE8_MEMBER( myb3k_state::myb3k_6845_address_w )
-{
-	LOGCRT("%s: %02x\n", FUNCNAME, data);
-
-	m_crtc_index = data;
-	m_crtc->address_w(space, offset, data);
 }
 
-WRITE8_MEMBER( myb3k_state::myb3k_6845_data_w )
-{
-	LOGCRT("%s: %02x\n", FUNCNAME, data);
-
-	m_crtc_vreg[m_crtc_index] = data;
-	m_crtc->register_w(space, offset, data);
-}
+/*
+ * Setup of 6845 in different graphics modes from basica command line
+ *  screen chars per row 36/40/80, rows 20/25, mode 0/1/2/3
+ *
+ *  chars 36  40  80  80  36  40  80  36  40
+ *  rows  25  25  25  25  20  20  20  25  25
+ *  mode   1   1   2   3   0   0   0   0   0 - as per basica "screen x,y,mode"
+ *----------------------------------------------------------------------------
+ *  R0    55  55  55  55  55  55  55  55  55 - Horizontal Total
+ *  R1    40  40  40  40  40  40  40  40  40 - Horizontal Displayed
+ *  R2    44  44  44  44  44  44  44  44  44 - Horizontal Sync Position
+ *  R3   132 132 132  52 132 132 132  52  52 - Sync Width
+ *  R4    31  31  31  26  24  24  24  26  26 - Vertical Total
+ *  R5     0   0   0   7   9   9   9   7   7 - Vertical Total Adjust
+ *  R6    25  25  25  25  20  20  20  25  25 - Vertical Displayed
+ *  R7    27  27  27  25  22  22  22  25  25 - Vertical Sync Position
+ *  R8     0   0   0   3   0   0   0   3   3 - Interlace & Skew
+ *  R9     7   7   7  14   9   9   9  14  14 - Maximum Raster Address
+ *  R10   96  96  96  96  96  96  96  96  96 - Cursor Start Address
+ *  R11    7   7   7  15   9   9   9  15  15 - Cursor End Address
+ *  R12    0   0   0   0   0   0   0   0   0 - Start Address (H)
+ *  R13    0   0   0   0   0   0   0   0   0 - Start Address (L)
+ * ---------------------------------------------------------------------------
+ *  vmode  1   1   2   6   1   1   2   5   5 - as per 3 bits latch at I/O 0x04
+ *  xres 320 320 640 640 320 320 640 320 320
+ *  yres 200 200 200 400 200 200 200 400 400
+ *  char  40  40  80  80  40  40  80  40  40
+ */
 
 WRITE8_MEMBER( myb3k_state::myb3k_video_mode_w )
 {
-	LOGCRT("%s: %02x\n", FUNCNAME, data);
+	LOG("%s: %02x\n", FUNCNAME, data);
+	LOGVMOD("Video Mode %02x\n", data);
 
 	/* ---- -x-- interlace mode */
 	/* ---- --xx horizontal step count (number of offsets of vram RAM data to skip, 64 >> n) */
 
 	m_vmode = data;
+	switch (data & 7)
+	{
+	case 0:
+		if ((ioport("MONITOR")->read() & 1) == 0)
+		{
+			LOGVMOD(" - Color display 320x200 on 40x25 8 grey tones\n");
+			rectangle rect(0, 320 - 1, 0, 200 - 1);
+			machine().first_screen()->configure(320, 200, rect, HZ_TO_ATTOSECONDS(50));
```

Comment:
`first_screen()` shouldn't be used in new code.

Reply:
Yeah, use `required_device` to find it relative to the root.  `first_screen` is going to go away at some point.